src/matrix.rs

```diff
@@ -0,0 +1,104 @@
+//! Collection of matrix implementations. 
+
+use std::str::FromStr;
+use std;
+use util;
+
+type AdjacencyMap <T> = util::HashMap<std::string::String, std::option::Option<T>>;
+type AdjacencyMatrix <T> = util::HashMap<std::string::String, AdjacencyMap<T>>;
+
+/// Adjacency matrix struct.
+pub struct Adjacency <M: Clone> {
+    edges : AdjacencyMatrix<M>,
+}
+
+///  Poor man's adjacency matrix biased towards incident edge queries.
+///
+///  Edges are not symmetric.  Two values are symmetrically adjacent when
+///  edges originate from each value to the other value.
+impl <M: Clone> Adjacency <M> {
+
+    /// Construct a new adjacency matrix.
+    pub fn new() -> Self {
+        Adjacency {
+            edges: Default::default(), 
+        }
+    } 
+
+    /// Adds an outbound edge from a node to another.
+    pub fn add_asymmetric_edge(&mut self, from_str: &str, to_str: &str, metadata: std::option::Option<M>) {
+        let to = std::string::String::from_str(to_str).unwrap();
```

Comment:
I _think_ this can be trimmed down to `String::from_str`.

Reply:
Fixed, and yes it can!